x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/ExplainDataFrameAnalyticsAction.java

```diff
@@ -30,6 +45,118 @@ private ExplainDataFrameAnalyticsAction() {
         super(NAME, ExplainDataFrameAnalyticsAction.Response::new);
     }
 
+    public static class Request extends AcknowledgedRequest<Request> implements ToXContentObject {
```

Comment:
I added this request object (just a copy of `PutDataFrameAnalytics`) as the PUT object was being reused here, but I didn't want to make that request's task cancellable (as PUT isn't cancellable).

Reply:
Also, I added a BWC test for `explain` in a mixed cluster to ensure this doesn't break anything. Ran it many many times and its all good since the explain request and the put request can be serialized to each other without issue.